tracer/src/Datadog.Trace.Tools.Runner/CheckAgentCommand.cs

```diff
@@ -0,0 +1,54 @@
+// <copyright file="CheckAgentCommand.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#nullable enable
+
+using System;
+using System.Threading.Tasks;
+using Datadog.Trace.Agent;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Tools.Runner.Checks;
+using Spectre.Console;
+using Spectre.Console.Cli;
+
+namespace Datadog.Trace.Tools.Runner
+{
+    internal class CheckAgentCommand : AsyncCommand<CheckAgentSettings>
+    {
+        public override async Task<int> ExecuteAsync(CommandContext context, CheckAgentSettings settings)
+        {
+            ExporterSettings configuration;
+
+            if (settings.Url == null)
+            {
+                // Try to autodetect the agent settings
+                configuration = new ExporterSettings(new EnvironmentConfigurationSource());
```

Comment:
Should we be using the full config source version here, i.e. including config files, web config etc?

Reply:
We're in the context of the tool, not of a target process/tool, so there's no such thing as config files.
When the agent connectivity check runs as part of the process checks `dd-trace check process <pid>` then it will use the config files in the resolution.

Reply:
Ah, good point 👍